src/test/java/com/rtb/projectmanagementtool/project/ProjectControllerTest.java

```diff
@@ -1,35 +1,136 @@
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
 import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.rtb.projectmanagementtool.project.ProjectController;
+import com.rtb.projectmanagementtool.project.ProjectData;
+import java.util.ArrayList;
+import java.util.Arrays;
 import org.junit.After;
+import org.junit.Assert;
 import org.junit.Before;
-import org.junit.Ignore;
+import org.junit.Test;
 
 public class ProjectControllerTest {
   private final LocalServiceTestHelper helper =
       new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  DatastoreService ds;
+  ProjectController projectController;
+
+  // Project Creators
+  private final Long USER1 = 1l;
+  private final Long USER2 = 2l;
+  private final Long USER3 = 3l;
+  private final Long USER4 = 4l;
+
+  private ProjectData PROJECT1 = new ProjectData("Project 1", "Project 1 Description", USER1);
+  private ProjectData PROJECT2 = new ProjectData("Project 2", "Project 2 Description", USER1);
+  private ProjectData PROJECT3 = new ProjectData("Project 3", "Project 3 Description", USER2);
+  private ProjectData PROJECT4 = new ProjectData("Project 4", "Project 4 Description", USER3);
 
   @Before
   public void setUp() {
     helper.setUp();
+
+    ds = DatastoreServiceFactory.getDatastoreService();
+    projectController = new ProjectController(ds);
+
+    // Add users to projects
+    PROJECT2.addAdminUser(USER2);
+    PROJECT2.addRegularUser(USER4);
+    PROJECT2.addAdminUser(USER3);
+    PROJECT1.addAdminUser(USER4);
+    PROJECT4.addRegularUser(USER2);
+    PROJECT3.addRegularUser(USER3);
+    PROJECT3.addRegularUser(USER1);
+
+    // Set hardcoded ids to mimic expected behavior
+    PROJECT1.setId(1l);
+    PROJECT2.setId(2l);
+    PROJECT3.setId(3l);
+    PROJECT4.setId(4l);
   }
 
   @After
   public void tearDown() {
     helper.tearDown();
   }
 
-  @Ignore
-  public void noProjects() {}
+  // One test method because I need tests to run in this order
+  @Test
+  public void runProjectControllerTest() {
+    testAddProjectsToDatabase();
+    testSaveProjectsToDatabase();
+    testGetProjectWithUserAndCreator();
+    testRemoveProjects();
+  }
```

Comment:
It would probably be better to keep the tests split up though because unit tests are supposed to be independent of one another. Maybe, for the tests that involve only retrieving some info from datastore, you can manually add entitities to datastore in that unit test without using projectController to add them? That way, you're only testing one function of projectController at a time.

Reply:
+1.  tests requiring a specific order is usually discouraged.  it'll be hard to debug if the test fails mid way since you can't easily tell if the current test has problems, or if one of the earlier tests provided an incorrect state.

Also as a side note, it's advisable to "test behaviors and not methods/functions", so we don't necessarily need 1 test per function.  Most of the time they're synonymous if the method only performs a single behavior.  But for cases where a method does multiple behaviors, it would be good to have multiple tests for this same method (one per distinct behavior).  Or cases where multiple functions are used to perform 1 full action, it would be better to test all those methods in conjunction.